asf_core_data/pipeline/data_joining/merge_install_dates.py

```diff
@@ -319,4 +282,9 @@ def manage_hp_install_dates(df, identifier="UPRN", verbose=False):
 
     df = pd.concat([df, no_future_hp_entry])
 
+    # Deduplicate and only keep latest record
+    df = df.sort_values("INSPECTION_DATE", ascending=True).drop_duplicates(
+        subset=[identifier], keep="last"
+    )
```

Comment:
doesn't this depend on the merged version we're considering? if all epc records, or only most relevant, etc etc?

Reply:
This data here is based on EPC, not the merged EPC/MCS, so these versions don't exist. We need the processed but undedupl. EPC data for extracting the best HP installation date based on the first HP mention in all EPC records.

This means that the EPC data going out of this function will have no duplicates but only keep the most recent record. For the purpose of the HPMT, I think this is a good option as this is the best appromixation for the current UK housing stock. 

If you want to analyse what the housing situation was right before/after installing a heat pump, you may need to make some adjustments but I think that's outside the scope of the HPMT, and we can easily add this if ever needed.

I'll leave a note that this can be adapted.

Reply:
yeah, I guess it might depend on use cases, but as you say it can easily be adapted :)